superduperdb/backends/ibis/data_backend.py

```diff
@@ -35,12 +61,24 @@ class IbisDataBackend(BaseDataBackend):
 
     db_type = DBType.SQL
 
-    def __init__(self, conn: BaseBackend, name: str, in_memory: bool = False):
-        super().__init__(conn=conn, name=name)
+    def __init__(self, uri: str, flavour: t.Optional[str] = None):
+        conn, name, in_memory = _IbisConnectionBuilder.build(uri, flavour)
+        super().__init__(uri=uri, flavour=flavour)
+        self.conn = conn
+        self.name = name
         self.in_memory = in_memory
+        self._setup(conn)
+
+    def _setup(self, conn):
         self.dialect = getattr(conn, 'name', 'base')
         self.db_helper = get_db_helper(self.dialect)
 
+    def reconnect(self):
+        # Reconnect to database.
+        conn, _, _ = _IbisConnectionBuilder.build(self.uri, self.flavour)
```

Comment:
Don't cache `self.uri`.

Reply:
Idea would be to build a callback at initialization time, and then keep that there.

Reply:
done